modules/auxiliary/server/capture/imap.rb

```diff
@@ -87,6 +113,12 @@ def on_client_data(c)
       return
     end
 
+    if cmd.upcase == 'ID'
+      c.put("* ID\r\n")
+      c.put("#{num} OK ID completed\r\n")
+      return
```

Comment:
Just checking here but from your WireShark capture it seems like ID had some parameters (unfortunately can't copy as its a screenshot but should be visible from original post). Do these not need to be sent here?

Reply:
They certainly aren't required, but maybe it should? RFC2971 says:

```
3.1. ID Command

   Arguments:  client parameter list or NIL

   Responses:  OPTIONAL untagged response: ID

   Result:     OK    identification information accepted
               BAD   command unknown or arguments invalid

   Implementation identification information is sent by the client with
   the ID command.

   This command is valid in any state.

   The information sent is in the form of a list of field/value pairs.
   Fields are permitted to be any IMAP4 string, and values are permitted
   to be any IMAP4 string or NIL.  A value of NIL indicates that the
   client can not or will not specify this information.  The client may
   also send NIL instead of the list, indicating that it wants to send
   no information, but would still accept a server response.

   The available fields are defined in section 3.3.

   Example:  C: a023 ID ("name" "sodr" "version" "19.34" "vendor"
                 "Pink Floyd Music Limited")
             S: * ID NIL
             S: a023 OK ID completed
```

So probably we could just do `NIL` to behave correclty

Reply:
"A value of NIL indicates that the client can not or will not specify this information." <- this seems reasonable explanation here only concern might be if the server is expecting the client to respond and will act differently if it doesn't send this information but tbh considering we probably won't know the required response most of the time I imagine this is acceptable.